File: mvvmapp/src/main/java/com/example/mvvmapp/base/BaseActivity.java
package com.example.mvvmapp.base;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;

/**
 * Author: Zeus
 * Date: 2020/7/13 17:07
 * Description:
 * History:
 */
public abstract class BaseActivity<D_T> extends AppCompatActivity {
    protected D_T mDataBinding;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ViewDataBinding viewDataBinding = DataBindingUtil.setContentView(this, getLayoutID());
        this.mDataBinding = (D_T) viewDataBinding;
        initData();
        loadData();
    }

    protected abstract void initData();

    protected abstract void loadData();

    protected abstract int getLayoutID();
}

File: app/src/main/java/com/kky/example/base/activity/BaseActivity.kt
package com.kky.example.base.activity

import android.content.Intent
import android.os.Bundle
import android.os.PersistableBundle
import androidx.appcompat.app.AppCompatActivity
import com.kky.example.util.LocalPrinter

/**
 * @name DemosSet
 * @author Zeus QQ:1946690608
 * @time 2018/4/8 19:58
 * @change time
 * @class describe
 */
open class BaseActivity : AppCompatActivity() {

    protected open var TAG: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        TAG = this.javaClass.simpleName
        LocalPrinter.printLocal("$TAG   onCreate")

    }


    override fun onDestroy() {
        super.onDestroy()
        LocalPrinter.printLocal("$TAG   onDestroy")
    }

    override fun onStop() {
        super.onStop()
        LocalPrinter.printLocal("$TAG   onStop")
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        LocalPrinter.printLocal("$TAG   onNewIntent")
    }
}